src/store/create-invitation/saga.ts

```diff
@@ -33,16 +33,24 @@ export function* fetchInvite() {
   }
 }
 
-export function* saga() {
-  const channel = yield call(getAuthChannel);
-
+function* listenForUserLogin() {
+  const authChannel = yield call(getAuthChannel);
   while (true) {
-    const { userId = undefined } = yield take(channel, '*');
+    yield take(authChannel, Events.UserLogin);
+    yield spawn(fetchInvite);
+  }
+}
 
-    if (userId) {
-      yield spawn(fetchInvite);
-    } else {
-      yield put({ type: SagaActionTypes.Cancel });
-    }
+function* listenForUserLogout() {
+  const authChannel = yield call(getAuthChannel);
+  while (true) {
+    yield take(authChannel, Events.UserLogout);
+    yield put({ type: SagaActionTypes.Cancel });
   }
 }
+
+export function* saga() {
+  yield spawn(listenForUserLogin);
+  yield spawn(listenForUserLogout);
+  yield takeEvery(SagaActionTypes.GetCode, fetchInvite);
```

Comment:
@dalefukami  I wanted to use `takeLatest` here (i.e take the Latest one, and ignore an already running saga action). BUT that doesn't work. I don't seem to understand fully why. The `fetchInvite` gets called yes, but it gets stuck on this line 

https://github.com/zer0-os/zOS/blob/aa448e2603e6683ea9ebedfb01ebffdfa9a91ed6/src/store/create-invitation/saga.ts#L19

i.e I don't receive any value of `invitation` (BUT the api gets called :/). 

And everything works as normal while using `takeEvery`..

Reply:
So, this is probably because there's a conflict here with `takeLatest` and the `while(true)` (line 8) in  the previous implementation. You're now listening for the `GetCode` even twice. Once in the `takeEvery` and then again at the start of the while loop. Since we can switch to the simpler takeEvery we no longer need the while loop.